contracts/ERC1155NonTransferable.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+pragma solidity 0.7.1;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155Pausable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./interfaces/IERC1155NonTransferable.sol";
+
+/**
+ * @title Non transferable token contract, implementing ERC-1155, but preventing transfers
+ */
+contract ERC1155NonTransferable is
+    IERC1155NonTransferable,
+    ERC1155Pausable,
+    Ownable
+{
+    constructor(string memory uri_) ERC1155(uri_) Ownable() {}
+
+    /*
+     * @notice Mint an amount of a desired token
+     * @dev ERC-1155
+     * @param _to       owner of the minted token
+     * @param _tokenId  ID of the token to be minted
+     * @param _value    Amount of the token to be minted
+     * @param _data     Additional data forwarded to onERC1155BatchReceived if _to is a contract
+     */
+    function mint(
+        address _to,
+        uint256 _tokenId,
+        uint256 _value,
+        bytes memory _data
+    ) public override onlyOwner {
+        _mint(_to, _tokenId, _value, _data);
+    }
+
+    /**
+     * @notice Batch minting of tokens
+     * @dev ERC-1155
+     * @param _to The address that will own the minted token
+     * @param _tokenIds IDs of the tokens to be minted
+     * @param _values Amounts of the tokens to be minted
+     * @param _data Additional data forwarded to onERC1155BatchReceived if _to is a contract
+     */
+    function mintBatch(
+        address _to,
+        uint256[] memory _tokenIds,
+        uint256[] memory _values,
+        bytes memory _data
+    ) public onlyOwner {
+        _mintBatch(_to, _tokenIds, _values, _data);
+    }
+
+    /**
+     * @notice Burn an amount of tokens with the given ID
+     * @dev ERC-1155
+     * @param _account  Account which owns the token
+     * @param _tokenId  ID of the token
+     * @param _value    Amount of the token
+     */
+    function burn(
+        address _account,
+        uint256 _tokenId,
+        uint256 _value
+    ) public override onlyOwner {
+        _burn(_account, _tokenId, _value);
```

Comment:
I am not sure I understand this, where is the _burn coming from ?

As we dont inherit from https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v3.4.2-solc-0.7/contracts/token/ERC1155/ERC1155Burnable.sol  ?

I must be being stupid, but it looks like the import isn't explicit in the code, which I find odd.

Reply:
We inherit `ERC1155Pausable` which inherit `ERC1155` which already implements internal `_burn`

`ERC1155Burnable` on the other hands implements public methods `burn` and `burnBatch` which are slighlty different from our impementation